src/Npgsql/Replication/PgOutput/PgOutputAsyncEnumerable.cs

```diff
@@ -77,52 +86,75 @@ await foreach (var xLogData in stream.WithCancellation(cancellationToken))
                 case BackendReplicationMessageCode.Begin:
                 {
                     await buf.EnsureAsync(20);
-                    yield return _beginMessage.Populate(
-                        xLogData.WalStart,
-                        xLogData.WalEnd,
-                        xLogData.ServerClock,
-                        new NpgsqlLogSequenceNumber(buf.ReadUInt64()),
-                        TimestampHandler.FromPostgresTimestamp(buf.ReadInt64()),
-                        buf.ReadUInt32()
-                    );
+                    yield return _beginMessage.Populate(xLogData.WalStart, xLogData.WalEnd, xLogData.ServerClock,
+                        transactionFinalLsn: new NpgsqlLogSequenceNumber(buf.ReadUInt64()),
+                        transactionCommitTimestamp: TimestampHandler.FromPostgresTimestamp(buf.ReadInt64()),
+                        transactionXid: buf.ReadUInt32());
+                    continue;
+                }
+                case BackendReplicationMessageCode.Message:
+                {
+                    uint? transactionXid;
+                    if (inStreamingTransaction)
+                    {
+                        await buf.EnsureAsync(14);
+                        transactionXid = buf.ReadUInt32();
+                    }
+                    else
+                    {
+                        await buf.EnsureAsync(10);
+                        transactionXid = null;
+                    }
+
+                    var flags = buf.ReadByte();
+                    var messageLsn = new NpgsqlLogSequenceNumber(buf.ReadUInt64());
+                    var prefix = await buf.ReadNullTerminatedString(async: true, cancellationToken);
+                    await buf.EnsureAsync(4);
+                    var length = buf.ReadUInt32();
+                    var data = (NpgsqlReadBuffer.ColumnStream)xLogData.Data;
+                    data.Init(checked((int)length), false);
+                    yield return _logicalDecodingMessage.Populate(xLogData.WalStart, xLogData.WalEnd, xLogData.ServerClock, transactionXid,
+                        flags, messageLsn, prefix, data);
                     continue;
                 }
                 case BackendReplicationMessageCode.Commit:
                 {
                     await buf.EnsureAsync(25);
-                    yield return _commitMessage.Populate(
-                        xLogData.WalStart,
-                        xLogData.WalEnd,
-                        xLogData.ServerClock,
-                        buf.ReadByte(),
-                        new NpgsqlLogSequenceNumber(buf.ReadUInt64()),
-                        new NpgsqlLogSequenceNumber(buf.ReadUInt64()),
-                        TimestampHandler.FromPostgresTimestamp(buf.ReadInt64())
-                    );
+                    yield return _commitMessage.Populate(xLogData.WalStart, xLogData.WalEnd, xLogData.ServerClock, buf.ReadByte(),
```

Comment:
I've gotten used to putting each parameter on its own line, but that's probably EF Core's fault :rofl:

Reply:
You know, I have a pretty big screen in landscape mode. I also know your desktop setup, so there might be an inherent conflict 😆.

Reply:
Yep :)